### PR TITLE
docment `Callback` core (13_callback.core)

### LINKED ISSUE #34: 13_callback.core: create docments
Add docments to [13_callback.core](https://github.com/muellerzr/fastai-docment-sprint/blob/master/nbs/13_callback.core.ipynb).

See the style guide and contribution guide for more details (links will be added when created).

This requires moderate to advanced familiarity with the fastai training loop and callbacks.

Docments progress:

- [x] Callback
- [x] TrainEvalCallback
- [x] Gather and fetch preds callbacks

### PR DESCRIPTION
**Pull request by Bharat123rox**
For #34 
Tagging @muellerzr to identify appropriate owners

### CODE REVIEW

**Comment by Bharat123rox** [comment]
Would like you to review `FetchPredsCallback` and provide inputs, I made best guesses based on the actual code of `get_preds()` but I'm not 100% sure, so would love the inputs from maintainers

File: fastai/callback/core.py [hunk 1]
```diff
@@ -105,8 +105,16 @@ def before_validate(self):
...
+    def __init__(self,
+    with_input:bool=False, # Whether to gather and return the inputs
+    with_loss:bool=False, # Whether to gather and return the losses
+    save_preds:(str, os.PathLike)=None, # Path to save the predictions
+    save_targs:(str, os.PathLike)=None, # Path to save the targets
+    with_preds:bool=True, # Whether to gather and return the predictions
+    with_targs:bool=True, # Whether to gather and return the targets
+    concat_dim:int=0, # The dimension on which to concatenate all the tensors
+    pickle_protocol:int=2 # The pickle protocol to use for saving the predictions and targets
+    ):
```
Comment (warner-benjamin) [inline fastai/callback/core.py:117]: Suggested changes to `GatherPredsCallback.__init__`
```suggestion
    def __init__(self,
        with_input:bool=False, # Whether to return inputs
        with_loss:bool=False, # Whether to return losses
        save_preds:Path=None, # Path to save predictions
        save_targs:Path=None, # Path to save targets
        with_preds:bool=True, # Whether to return predictions
        with_targs:bool=True, # Whether to return targets
        concat_dim:int=0, # Dimension to concatenate returned tensors
        pickle_protocol:int=2 # Pickle protocol used to save predictions and targets
    ):
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
Please update the docments to reflect the [style guide](https://github.com/muellerzr/fastai-docment-sprint/blob/master/docs_src/style-guide.md) formatting and rephrase the docment comments to be more concise.

I modified `GatherPredsCallback.__init__` to reflect those changes as a suggestion.

Also, could you add doc strings to the `GatherPredsCallback` and `FetchPredsCallback` methods which are missing doc strings?

**Comment by Bharat123rox** [comment]
Apologies @warner-benjamin I had some problems with commit metadata earlier and had to force push to remove those

File: fastai/callback/core.py [hunk 2]
```diff
@@ -151,11 +161,19 @@ def all_tensors(self):
...
     "A callback to fetch predictions during the training loop"
     remove_on_fetch = True
-    def __init__(self, ds_idx=1, dl=None, with_input=False, with_decoded=False, cbs=None, reorder=True):
+    def __init__(self,
+        ds_idx:int=1, # Index of databunchset in fetching `Learner` predictions if `dl` is not present
+        dl:DataLoader=None, # `DataLoader` in fetching `Learner` predictions
+        with_input:bool=False, # Whether to return inputs in `GatherPredsCallback`
+        with_decoded:bool=False, # Whether to return decoded predictions from `decodes` method of `Learner` loss function
+        cbs:list=None, # `Callback` list for the `Learner`
+        reorder:bool=True # Whether to sort prediction results
```
Comment (marii-moe) [inline fastai/callback/core.py:170]:
```suggestion
    def __init__(self,
        ds_idx:int=1, # Index of dataset, 0 for train, 1 for valid, used if `dl` is not present
        dl:DataLoader=None, # `DataLoader` used for fetching `Learner` predictions
        with_input:bool=False, # Whether to return inputs in `GatherPredsCallback`
        with_decoded:bool=False, # Whether to return predicted class
        cbs:list=None, # `Callback` list to add to the `Learner`
        reorder:bool=True # Whether to sort prediction results
```

**Review by marii-moe — CHANGES_REQUESTED** [review]

File: fastai/callback/core.py (hunk 2 above)
Comment (marii-moe) [reply fastai/callback/core.py:170]: Added a few clarifications to the docs.

**Review by marii-moe — APPROVED** [review]
LGTM

**Review by warner-benjamin — APPROVED** [review]

**Pull request merged**